gen-env.sh

```diff
@@ -0,0 +1,107 @@
+#!/bin/sh
+# generate .env file for testing up's bot
+#  https://github.com/unixporn/Supreme-Demolition-Droid
+#
+# server template link:
+#  https://discord.new/zkhTrUTEbtg9
+#
+# link to add bot to server:
+#  https://discord.com/oauth2/authorize?scope=bot&client_id=<BOTID>
+# where <BOTID> is the bot's snowflake
+
+[ "$token" ] || token=TOKENGOESHERE
+serverid=SERVERIDGOESHERE
+
+
+[ "$token" ] && [ "$serverid" ] || exec \
+	echo "Please add the bot's token and the server's ID to the script"
+
+# clean the env
+unset col start mod mute intern
+unset showcase modcat feedback modlog botmod botlog botstuff humantrafficking
+
+#
+#  roles
+#
+
+# get general server info
+# -> use jq to get the roles -> remove quotes -> reverse list
+roles=$(curl -s -X GET \
+	-H "Authorization: Bot $token" \
+	-H "Content-Type: application/json" \
+	"https://discord.com/api/v9/guilds/$serverid" \
+	| jq '.roles[]' -c | tr -d '"' | tac)
+
+# do a line-by-line loop of the roles to get the variables
+while IFS=':,' read -r _ id _ name _; do
+	case $name in
+		"@everyone")       ;;
+		"mods")            mod=$id;;
+		"unpaid intern")   intern=$id;;
+		"mute")            mute=$id;;
+		# colours
+		"black")           col="$id" start=1;;
+		*) [ "$start" ] && col="$col,$id";;
+	esac
+done << EOF
+$roles
+EOF
+
+
+#
+#  channels
+#
+
+# get channel list -> use jq to put every channel on a new line -> delete quotes
+channels=$(curl -s -X GET \
+	-H "Authorization: Bot $token" \
+	-H "Content-Type: application/json" \
+	"https://discord.com/api/v9/guilds/$serverid/channels" | jq '.[]' -c | tr -d '"')
```

Comment:
more future proof, in case order ever changes, which is likely

```suggestion
	"https://discord.com/api/v9/guilds/$serverid/channels" | jq -r '.[] | [.id, .last_message_id, .type, .name] | join("||")'')
```

Reply:
Thanks!  

lastmessageid isn't needed, but jq's syntax is bad